compiler/rustc_span/src/span_encoding.rs

```diff
@@ -372,9 +379,9 @@ impl Span {
     pub fn eq_ctxt(self, other: Span) -> bool {
         match (self.inline_ctxt(), other.inline_ctxt()) {
             (Ok(ctxt1), Ok(ctxt2)) => ctxt1 == ctxt2,
-            (Ok(ctxt), Err(index)) | (Err(index), Ok(ctxt)) => {
-                with_span_interner(|interner| ctxt == interner.spans[index].ctxt)
-            }
+            // Ok: ctxt <= MAX_CTXT
+            // Err: ctxt > MAX_CTXT
```

Comment:
I think this change makes sense but can't work out what this comment means.

Reply:
I have reworded it